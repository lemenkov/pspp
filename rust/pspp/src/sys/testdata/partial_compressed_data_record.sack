# File header.
"$FL2"; s60 "$(#) SPSS DATA FILE PSPP synthetic test file";
2; # Layout code
6; # Nominal case size
1; # Compressed
0; # Not weighted
-1; # Unspecified number of cases.
100.0; # Bias.
"01 Jan 11"; "20:53:52"; s64 "PSPP synthetic test file";
i8 0 *3;

# Numeric variables.
2; 0; 0; 0; 0x050800 *2; s8 "NUM1";
2; 0; 0; 0; 0x050800 *2; s8 "NUM2";

# String variable.
2; 4; 0; 0; 0x010400 *2; s8 "STR4";
2; 8; 0; 0; 0x010800 *2; s8 "STR8";
2; 15; 0; 0; 0x010f00 *2; s8 "STR15";
2; -1; 0; 0; 0; 0; s8 "";

# Character encoding record.
7; 20; 1; 12; "windows-1252";

# Dictionary termination record.
999; 0;

# Compressed data.
i8 1 100 254 253 254 253; i8 255 251; "abcdefgh"; s8 "0123";
