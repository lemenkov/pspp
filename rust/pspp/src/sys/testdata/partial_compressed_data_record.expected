╭──────────────────────┬────────────────────────╮
│       Created        │    01-JAN-2011 20:53:52│
├──────────────────────┼────────────────────────┤
│Writer Product        │PSPP synthetic test file│
├──────────────────────┼────────────────────────┤
│       Compression    │SAV                     │
│       Number of Cases│Unknown                 │
╰──────────────────────┴────────────────────────╯

╭─────────┬────────────────────────╮
│Label    │PSPP synthetic test file│
│Variables│                       5│
╰─────────┴────────────────────────╯

╭─────┬────────┬─────┬─────────────────┬─────┬─────┬─────────┬────────────┬────────────┬──────────────╮
│     │Position│Label│Measurement Level│ Role│Width│Alignment│Print Format│Write Format│Missing Values│
├─────┼────────┼─────┼─────────────────┼─────┼─────┼─────────┼────────────┼────────────┼──────────────┤
│num1 │       1│     │                 │Input│    8│Right    │F8.0        │F8.0        │              │
│num2 │       2│     │                 │Input│    8│Right    │F8.0        │F8.0        │              │
│str4 │       3│     │Nominal          │Input│    4│Left     │A4          │A4          │              │
│str8 │       4│     │Nominal          │Input│    8│Left     │A8          │A8          │              │
│str15│       5│     │Nominal          │Input│   15│Left     │A15         │A15         │              │
╰─────┴────────┴─────┴─────────────────┴─────┴─────┴─────────┴────────────┴────────────┴──────────────╯

Unexpected end of file at offset 0x1ac, 0 bytes and 2 compression chunks into a compressed case.

╭────┬──────┬────┬────────┬────────────────┬──────────────────────────────╮
│Case│ num1 │num2│  str4  │      str8      │             str15            │
├────┼──────┼────┼────────┼────────────────┼──────────────────────────────┤
│1   │-99.00│ .00│        │abcdefgh        │        0123                  │
╰────┴──────┴────┴────────┴────────────────┴──────────────────────────────╯
